cmd/kubelet/app/server.go

```diff
@@ -597,6 +536,48 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func getCloud(s *options.KubeletServer) (cloud cloudprovider.Interface, err error) {
+	if !cloudprovider.IsExternal(s.CloudProvider) && s.CloudProvider != componentconfigv1alpha1.AutoDetectCloudProvider {
+		cloud, err = cloudprovider.InitCloudProvider(s.CloudProvider, s.CloudConfigFile)
```

Comment:
what happens if InitCloudProvider is called multiple times? making this a function encourages that